samples/TagHelperSample.Web/TagHelpers/ConditionalCommentTagHelper.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+
+namespace TagHelperSample.Web
+{
+    [TargetElement("iecondition")]
+    public class ConditionalCommentTagHelper : TagHelper
+    {
+        [HtmlAttributeName("mode")]
+        public string Mode { get; set; }
+
+        [HtmlAttributeName("condition")]
+        public string Condition { get; set; }
+
+        public override void Process(TagHelperContext context, TagHelperOutput output)
+        {
+            output.TagName = null;
+            string startCondition = "";
+            string endCondition = "";
+            if (Mode == "downlevelHidden")
+            {
+                startCondition = $"<!--[if {Condition}]>";
+                endCondition = "<![endif]-->";
+
+            }
+            else if (Mode == "downlevelRevealed")
+            {
+                startCondition = $"<![if {Condition}]>";
+                endCondition = "<![endif]>";
+            }
```

Comment:
Not easy to tell the difference between what the `if` and `else` blocks do. If `Mode` remains (seeing @NTaylorMullen's recent [comment](https://github.com/aspnet/Mvc/pull/3007#discussion-diff-37492621)), suggest instead storing `string.Empty` or `"--"` in a variable and using that in the string concatenation / `Append()` expressions.